### PR TITLE
Blog App: validations and model specs

### PR DESCRIPTION
**Pull request by batoorsohail**
### In this project we:

- [x] Added the following validations:
   - For the User model:
      - Name must not be blank.
      - PostsCounter must be an integer greater than or equal to zero.
   - For the Post model:
      - The title must not be blank.
      - The title must not exceed 250 characters.
      - CommentsCounter must be an integer greater than or equal to zero.
      - LikesCounter must be an integer greater than or equal to zero.
   
- [x] Added unit specs for all of your models' methods and validations.

### CODE REVIEW

File: spec/comment_spec.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  end
+
+  it 'Reutrns post id' do
+    expect(comment.post_id).to eql 1
+  end
+
+  it 'checks comments_counter method' do
+    User.new(name: 'Ahmad', photo: 'www.iamge.com', bio: 'Full Stack Developer', posts_counter: 5)
+    post = Post.new(author_id: 1, title: 'New', text: 'First post', comments_counter: 3, likes_counter: 2)
+    comment = Comment.new(author_id: 1, post:)
```
Comment (Whoistolu) [inline spec/comment_spec.rb]: - [ ] [REQUIRED I currently get a syntax error from this file while trying to test. Kindly ensure that you provide a value for the post-argument. By providing a valid value for the post-argument, the syntax error should be resolved. 

Screenshot 2023-05-19 at 17 12 14

**Review by Whoistolu — CHANGES_REQUESTED** [review]
Hi @batoorsohail 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️

- [ ]  _Check the inline comments under the review for sections to improve._

### Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better.

### N/A

Cheers, and Happy coding!

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by rustam-tolipov — APPROVED** [review]
Hi @batoorsohail, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Developer-512px-10

## Highlights
1. No Linters errors 💯
2. All requirements are complete 💯
3. Well-implemented code 👍 

## Optional suggestions
No optional suggestions :)

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by batoorsohail** [comment]
Thanks @Whoistolu and @Rustamxon7 😄 🙌

**Pull request merged**